smac/epm/base_gp.py

```diff
@@ -9,19 +11,55 @@ def __init__(self, types, bounds, seed, **kwargs):
         """
         Abstract base class for all Gaussian process models.
         """
-        super().__init__(types=types, bounds=bounds, seed=seed,
-                         **kwargs)
+        super().__init__(types=types, bounds=bounds, seed=seed, **kwargs)
 
         self.rng = np.random.RandomState(seed)
 
-        lower = []
-        upper = []
-        for bound in bounds:
-            lower.append(bound[0])
-            upper.append(bound[1])
+    def _normalize_y(self, y):
+        self.mean_y_ = np.mean(y)
+        self.std_y_ = np.std(y)
+        if self.std_y_ == 0:
+            self.std_y_ = 1
```

Comment:
This requires explanation

Reply:
The idea is to avoid a divide by zero. If the std is zero, this implies that all samples are the same, which needs to be taken into account. This behavior is similar to what scikit-learn does.